jenkins/common.groovy

```diff
@@ -0,0 +1,180 @@
+
+
+VETSGOV_BUILDTYPES = [
+  'development',
+  'staging',
+  'production'
+]
+
+VAGOV_BUILDTYPES = [
+  'vagovdev',
+  'vagovstaging',
+  'vagovprod'
+]
+
+DRUPAL_MAPPING = [
+  'dev': 'vagovdev',
+  'staging': 'vagovstaging',
+  'live': 'vagovprod',
+]
+
+DEV_BRANCH = 'master'
+STAGING_BRANCH = 'master'
+PROD_BRANCH = 'master'
+
+IS_DEV_BRANCH = env.BRANCH_NAME == DEV_BRANCH
+IS_STAGING_BRANCH = env.BRANCH_NAME == STAGING_BRANCH
+IS_PROD_BRANCH = env.BRANCH_NAME == PROD_BRANCH
+
+def isReviewable() {
+  return !IS_DEV_BRANCH && !IS_STAGING_BRANCH && !IS_PROD_BRANCH
+}
+
+def isDeployable() {
+  return (IS_DEV_BRANCH ||
+          IS_STAGING_BRANCH) &&
+    !env.CHANGE_TARGET &&
+    !currentBuild.nextBuild // if there's a later build on this job (branch), don't deploy
+}
+
+def shouldBail() {
+  // abort the job if we're not on deployable branch (usually master) and there's a newer build going now
+  return !IS_DEV_BRANCH &&
+    !IS_STAGING_BRANCH &&
+    !IS_PROD_BRANCH &&
+    !env.CHANGE_TARGET &&
+    currentBuild.nextBuild
+}
+
+def runDeploy(jobName, ref) {
+  build job: jobName, parameters: [
+    booleanParam(name: 'notify_slack', value: true),
+    stringParam(name: 'ref', value: ref),
+  ], wait: false
+}
+
+def buildDetails(String buildtype, String ref) {
+  return """\
+BUILDTYPE=${buildtype}
+NODE_ENV=production
+BRANCH_NAME=${env.BRANCH_NAME}
+CHANGE_TARGET=${env.CHANGE_TARGET}
+BUILD_ID=${env.BUILD_ID}
+BUILD_NUMBER=${env.BUILD_NUMBER}
+REF=${ref}
+"""
+}
+
+def slackNotify() {
+  if (IS_DEV_BRANCH || IS_STAGING_BRANCH || IS_PROD_BRANCH) {
+    message = "vets-website ${env.BRANCH_NAME} branch CI failed. |${env.RUN_DISPLAY_URL}".stripMargin()
+    slackSend message: message,
+      color: 'danger',
+      failOnError: true
+  }
+}
+
+def setup(String dockerTag, String dockerArgs) {
+  stage("Setup") {
+
+    buildTypeOverride = DRUPAL_MAPPING.get(params.get('cmsEnv', 'none'), null)
+    if(buildTypeOverride) {
+      VAGOV_BUILDTYPES = [buildTypeOverride]
+    }
```

Comment:
Does 'buildTypeOverride' have to be applied here, or can it be applied at the top-level scope? Just thinking it might be clearer to initialize `VAGOV_BUILDTYPES` like -

```groovy
ALL_VAGOV_BUILDTYPES = 
  'vagovdev',
  'vagovstaging',
  'vagovprod'
]

BUILD_TYPE_OVERRIDE = DRUPAL_MAPPING.get(params.get('cmsEnv', 'none'), null)

VAGOV_BUILDTYPES = BUILD_TYPE_OVERRIDE ? [BUILD_TYPE_OVERRIDE]  : ALL_VAGOV_BUILDTYPES
```